src/main/java/snowy/common/FindCommand.java

```diff
@@ -0,0 +1,55 @@
+package snowy.common;
+
+import snowy.tasklist.Task;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Finds and lists all tasks in the task list whose description contains any of the argument keywords.
+ * Keyword matching is case-sensitive.
+ */
+public class FindCommand extends Command {
+
+    private final String keyword;
+
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    @Override
+    public CommandResult execute() {
+        final List<Task> tasksFound = getTasksWithKeywordInDescription(keyword);
+        return new CommandResult(getMessageForTaskListShownSummary(tasksFound));
+    }
+
+    /**
+     * Retrieves all tasks in the task list whose descriptions contain some of the specified keywords.
+     *
+     * @param keyword for searching
+     * @return list of tasks found
+     */
+    private List<Task> getTasksWithKeywordInDescription(String keyword) {
```

Comment:
I like that the method names are specific, but perhaps they can be a bit long and can be shortened?